example-apps/internal-knowledge-search/api/app.py

```diff
@@ -0,0 +1,185 @@
+from flask import Flask, jsonify, request, Response, current_app
+from flask_cors import CORS
+from elasticsearch_client import elasticsearch_client
+import os
+import sys
+import requests
+
+app = Flask(__name__, static_folder="../frontend/build", static_url_path="/")
+CORS(app)
+
+
+def get_identities_index(search_app_name):
+    search_app = elasticsearch_client.search_application.get(
+        name=search_app_name)
+    identities_indices = elasticsearch_client.indices.get(
+        index=".search-acl-filter*")
+    secured_index = [
+        app_index
+        for app_index in search_app["indices"]
+        if ".search-acl-filter-" + app_index in identities_indices
+    ]
+    if len(secured_index) > 0:
+        identities_index = ".search-acl-filter-" + secured_index[0]
+        return identities_index
+    else:
+        raise ValueError(
+            "Could not find identities index for search application %s", search_app_name
+        )
+
+
+@app.route("/")
+def api_index():
+    return app.send_static_file("index.html")
+
+
+@app.route("/api/default_settings", methods=["GET"])
+def default_settings():
+    return {
+        "elasticsearch_endpoint": os.getenv("ELASTICSEARCH_URL") or "http://localhost:9200"
+    }
+
+
+@app.route("/api/search_proxy/<path:text>", methods=["POST"])
+def search(text):
+    response = requests.request(
+        method="POST",
+        url=os.getenv("ELASTICSEARCH_URL") + '/' + text,
+        data=request.get_data(),
+        allow_redirects=False,
+        headers={"Authorization": request.headers.get(
+            "Authorization"), "Content-Type": "application/json"}
+    )
+
+    return response.content
+
+
+@app.route("/api/persona", methods=["GET"])
+def personas():
+    try:
+        search_app_name = request.args.get("app_name")
+        identities_index = get_identities_index(search_app_name)
+        response = elasticsearch_client.search(
+            index=identities_index, size=1000)
+        hits = response["hits"]["hits"]
+        personas = [x["_id"] for x in hits]
+        personas.append("admin")
+        return personas
+
+    except Exception as e:
+        current_app.logger.warn(
+            "Encountered error %s while fetching personas, returning default persona", e
+        )
+        return ["admin"]
+
+
+@app.route("/api/indices", methods=["GET"])
+def indices():
+    try:
+        search_app_name = request.args.get("app_name")
+        search_app = elasticsearch_client.search_application.get(
+            name=search_app_name)
+        return search_app['indices']
+
+    except Exception as e:
+        current_app.logger.warn(
+            "Encountered error %s while fetching personas, returning default persona", e
+        )
+        return ["admin"]
```

Comment:
This messaging looks like copy-pasta from above. Should probably just raise a 500 if you can't read the indices list from the search application?

Reply:
Done